docs: add section about services with long startup time (#1604)
canonical/operator: docs/howto/run-workloads-with-a-charm-kubernetes.md

Consider the K8s liveness success (`level=alive` check) to mean "Pebble is alive" rather than "the application is fully alive" (and failure to mean "this container needs to die"). For charms that take a long time to start, you should not have a `level=alive` check (if Pebble's running, it will report alive to K8s), and instead use an ordinary Pebble check (without a `level`) in conjunction with `on-check-failure: restart`. That way Pebble itself has full control over restarting the service in question.

### Services with long startup time

> When a K8s liveness probe (a `level-alive` check) succeeds, you should consider it to mean "Pebble is alive" rather than "workload is alive". Similarly, probe failure mean "this container needs to be restarted".

Reviewer: Some small suggestions/fixes:

```suggestion
When a K8s liveness probe (a `level=alive` check) succeeds, you should consider it to mean "Pebble is alive" rather than "the workload is alive". Similarly, a liveness probe failure means "this container needs to be restarted" rather than an issue with the workload.
```

> This means that for workloads that take a long or indefinite period of time to start, you should not have a `level=alive` check for that service. Instead, use an ordinary Pebble check without a level in conjunction with `on-check-failure: restart` on the service. That way Pebble itself has control over restarting the service in question.

Reviewer: I think it's fine to say "the service" instead of "that service" and "the service in question". Also I'd suggest adjusting the middle sentence slightly for readability:

```suggestion
This means that for workloads that take a long or indefinite period of time to start, you should not have a `level=alive` check for the service. Instead, use a Pebble check without a level and specify `on-check-failure: restart` for the service. That way Pebble itself has control over restarting the service.
```
